mediachain/reader/api.py

```diff
@@ -24,54 +24,31 @@ def get_object(transactor, object_id, fetch_images=True):
     except KeyError as e:
         pass
 
-    if fetch_images and obj.get('type') == 'artefact':
-        return fetch_thumbnails(obj)
-
     return obj
 
 
-def fetch_thumb_from_datastore(obj):
-    try:
-        ref = multihash_ref(obj['meta']['data']['thumbnail']['link'])
-        db = get_raw_datastore()
-        thumb = db.get(ref, timeout=10)
-        return thumb
-    except (ValueError, LookupError,
-            requests.exceptions.RequestException) as e:
-        # print 'error fetching from raw datastore: {}'.format(e)
+def open_binary_asset(asset, timeout=None):
+    if not asset.get('binary_asset', False):
         return None
 
-
-def fetch_thumb_from_uri(obj):
+    # try to load from ipfs if `link` is present
     try:
-        uri = obj['meta']['data']['thumbnail']['uri']
-        req = requests.get(uri, timeout=10)
-        return req.content
-    except (LookupError, requests.exceptions.RequestException) as e:
-        # print 'error fetching from uri: {}'.format(e)
-        return None
-
-
-def fetch_thumbnails(obj):
-    def with_fallback():
-        o = copy.deepcopy(obj)
-        if 'meta' not in o:
-            o['meta'] = {}
-        if 'data' not in o['meta']:
-            o['meta']['data'] = {}
-
-        o['meta']['data']['thumbnail_base64'] = 'NO_IMAGE'
-        return o
+        ref = multihash_ref(asset['link'])
+        db = get_raw_datastore()
+        return contextlib.closing(db.open(ref, timeout=timeout))
+    except (LookupError, ValueError, requests.exceptions.RequestException):
+        pass
 
-    thumb = fetch_thumb_from_datastore(obj)
-    if thumb is None:
-        thumb = fetch_thumb_from_uri(obj)
-    if thumb is None:
-        return with_fallback()
+    # If not, try to get the http url if present
+    try:
+        uri = asset['uri']
+        resp = requests.get(uri, stream=True, timeout=timeout)
```

Comment:
maybe not... this seemed like the simplest way to get a file-like object out of requests... But downloading up front is probably less fragile.  Especially if we want to hang on to the returned object for a while before reading from it; the remote end might close the connection before we can pull the data.

I'll change this to just wrap `resp.content` in a `BytesIO` or something.